src/metal.cs

```diff
@@ -1741,6 +2001,69 @@ partial interface MTLRenderCommandEncoder : MTLCommandEncoder {
 #endif
 		[Export ("drawIndexedPatches:patchIndexBuffer:patchIndexBufferOffset:controlPointIndexBuffer:controlPointIndexBufferOffset:indirectBuffer:indirectBufferOffset:")]
 		void DrawIndexedPatches (nuint numberOfPatchControlPoints, [NullAllowed] IMTLBuffer patchIndexBuffer, nuint patchIndexBufferOffset, IMTLBuffer controlPointIndexBuffer, nuint controlPointIndexBufferOffset, IMTLBuffer indirectBuffer, nuint indirectBufferOffset);
+
+		[Mac (10,13), NoiOS, NoTV, NoWatch]
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[Export ("setViewports:count:")]
+		void SetViewports (IntPtr viewports, nuint count);
```

Comment:
@timrisi is the app developer supposed to implement this protocol himself, or will this protocol only be used to call system-provided instances?

If only the system provides the instance, we don't necessarily care much about how it's bound, since user's won't have to override the method, and we can provide manually written extension methods with nicer API for calling the protocol.